api/v1/categories/PkpCategoryCategoryController.php

```diff
@@ -0,0 +1,409 @@
+<?php
+
+/**
+ * @file api/v1/categories/PkpCategoryCategoryController.php
+ *
+ * Copyright (c) 2014-2025 Simon Fraser University
+ * Copyright (c) 2003-2025 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class PkpCategoryCategoryController
+ *
+ * @ingroup api_v1_category
+ *
+ * @brief Handle API requests for category operations.
+ */
+
+namespace PKP\API\v1\categories;
+
+use APP\core\Application;
+use APP\facades\Repo;
+use APP\file\PublicFileManager;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Request;
+use Illuminate\Http\Response;
+use Illuminate\Support\Facades\Route;
+use PKP\category\Category;
+use PKP\components\forms\context\CategoryForm;
+use PKP\context\Context;
+use PKP\core\Core;
+use PKP\core\PKPBaseController;
+use PKP\core\PKPRequest;
+use PKP\file\TemporaryFile;
+use PKP\file\TemporaryFileManager;
+use PKP\security\authorization\CanAccessSettingsPolicy;
+use PKP\security\authorization\ContextAccessPolicy;
+use PKP\security\Role;
+
+class PkpCategoryCategoryController extends PKPBaseController
+{
+    /**
+     * @inheritDoc
+     */
+    public function authorize(PKPRequest $request, array &$args, array $roleAssignments): bool
+    {
+        $this->addPolicy(new ContextAccessPolicy($request, $roleAssignments));
+        $this->addPolicy(new CanAccessSettingsPolicy());
+
+        return parent::authorize($request, $args, $roleAssignments);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getHandlerPath(): string
+    {
+        return 'categories';
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getRouteGroupMiddleware(): array
+    {
+        return [
+            'has.user',
+            'has.context',
```

Comment:
Don't we have a non-stringified version of this? I think we should use less hardcoded strings.

```suggestion
            'has.user',
            'has.context',
```

Reply:
There are no non-stringified version for `'has.user'` & `'has.context'`, but I agree it would be good to eventually move away from repeatedly using these hardcoded strings